test/install_test.go

```diff
@@ -445,6 +475,111 @@ func TestRetrieveUidPostUpgrade(t *testing.T) {
 	}
 }
 
+type expectedData struct {
+	pod        string
+	cpuLimit   string
+	cpuRequest string
+	memLimit   string
+	memRequest string
+}
+
+var expectedResources = []expectedData{
+	{
+		pod:        "linkerd-controller",
+		cpuLimit:   "1010m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "101Mi",
+	},
+	{
+		pod:        "linkerd-destination",
+		cpuLimit:   "1020m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "102Mi",
+	},
+	{
+		pod:        "linkerd-grafana",
+		cpuLimit:   "1030m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "103Mi",
+	},
+	{
+		pod:        "linkerd-identity",
+		cpuLimit:   "1040m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "104Mi",
+	},
+	{
+		pod:        "linkerd-prometheus",
+		cpuLimit:   "1050m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "105Mi",
+	},
+	{
+		pod:        "linkerd-proxy-injector",
+		cpuLimit:   "1060m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "106Mi",
+	},
+	/*	 not used in default case
+	{
+		pod:        "linkerd-smi-metrics",
+		cpuLimit:   "1070m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "1007i",
+	},
+	*/
+	{
+		pod:        "linkerd-sp-validator",
+		cpuLimit:   "1080m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "108Mi",
+	},
+	{
+		pod:        "linkerd-tap",
+		cpuLimit:   "1090m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "109Mi",
+	},
+	{
+		pod:        "linkerd-web",
+		cpuLimit:   "1100m",
+		cpuRequest: "20m",
+		memLimit:   "200Mi",
+		memRequest: "110Mi",
+	},
+}
+
+func TestComponentProxyResources(t *testing.T) {
+	if TestHelper.UpgradeHelmFromVersion() == "" {
+		t.Skip("Skipping as this is not a helm upgrade test")
+	}
+
+	for _, expected := range expectedResources {
+		resourceReqs, err := TestHelper.GetResources("linkerd-proxy", expected.pod, TestHelper.GetLinkerdNamespace())
+		if err != nil {
+			t.Fatalf("Error retrieving resource requirements for %s: %s", expected.pod, err)
+		}
+
+		cpuLimitStr := resourceReqs.Limits.Cpu().String()
+		if cpuLimitStr != expected.cpuLimit {
+			t.Fatalf("unexpected %s CPU limit: expected %s, was %s", expected.pod, expected.cpuLimit, cpuLimitStr)
+		}
+		cpuRequestStr := resourceReqs.Requests.Cpu().String()
+		if cpuRequestStr != expected.cpuRequest {
+			t.Fatalf("unexpected %s CPU limit: expected %s, was %s", expected.pod, expected.cpuRequest, cpuRequestStr)
+		}
+	}
+}
```

Comment:
Nice tests :+1: 
But it appears you're not using `memLimit` nor `memRequest`?
Also, please replace the `t.Fatalf` calls with `testutil.AnnotatedFatalf(t, genericMsg, detailedMsg, args...)` that we're using now so that our CI reports can aggregate over these errors. `genericMsg` should be a message without details like "Unexpected CPU limit" and `detailedMsg`+`args` should be the detailed message like the ones you have.

Reply:
Added memory tests